megamek/src/megamek/common/MovePath.java

```diff
@@ -567,10 +567,10 @@ public void compile(final Game g, final Entity en, boolean clip) {
                 step = new MoveStep(this, step.getType(), step.getBraceLocation());
             } else if (!step.getLaunched().isEmpty()) {
                 step = new MoveStep(this, step.getType(), step.getLaunched());
+            } else if (step.isManeuver()) {
+                step = new MoveStep(this, step.getType(), step.hasNoCost(), step.isManeuver(), step.getManeuverType());
             } else if (step.getManeuverType() != ManeuverType.MAN_NONE) {
                 step = new MoveStep(this, step.getType(), -1, -1, step.getManeuverType());
-            } else if (step.isManeuver()) {
-                step = new MoveStep(this, step.getType(), step.hasNoCost(), step.isManeuver());
```

Comment:
This change in  order was required to preserve the noCost field of the additional steps of the maneuver. I thought the check for maneuverType could be removed along with the constructor it uses, but when I tried that all the maneuvers lost their thrust costs. I decided to leave well enough alone.